rados/ioctx.go

```diff
@@ -883,3 +883,77 @@ func (ioctx *IOContext) BreakLock(oid, name, client, cookie string) (int, error)
 		return int(ret), RadosError(int(ret))
 	}
 }
+
+// Snapshots
+// NOTE: self managed snapshots have not been implemented yet.
+
+// SnapHead represents the id of "no snapshot" used to resume normal operation when
+// reading and writing from a snapshot ioctx.
+const SnapHead uint64 = uint64(C.LIBRADOS_SNAP_HEAD)
+
+// SnapCreate creates a pool-wide snapshot
+func (ioctx *IOContext) SnapCreate(snapname string) error {
+	c_snapname := C.CString(snapname)
```

Comment:
Please use more go-idomatic naming (`cSnapName` not `c_snapname`) in all net new code, thanks! I won't repeat this for the remaining functions but it applies there too.